lib/Parser/BGParseManager.cpp

```diff
@@ -520,6 +552,51 @@ void BGParseWorkItem::ParseUTF8Core(Js::ScriptContext* scriptContext)
         Assert(this->cse.ei.bstrSource != nullptr);
         Assert(func == nullptr);
     }
+
+    LEAVE_PINNED_SCOPE();
+}
+
+// Deserializes the background parse results into this thread
+// Note: *must* run on a UI/Execution thread with an available ScriptContext
+HRESULT BGParseWorkItem::DeserializeParseResults(
+    Js::ScriptContext* scriptContextUI,
+    LPCUTF8 pszSrc,
+    SRCINFO const * pSrcInfo,
+    Js::Utf8SourceInfo* utf8SourceInfo,
+    Js::FunctionBody** functionBodyReturn,
+    size_t& srcLength,
+    uint& sourceIndex
+)
+{
+    HRESULT hr = this->parseHR;
+    if (hr == S_OK)
+    {
+        srcLength = this->parseSourceLength;
+        sourceIndex = scriptContextUI->SaveSourceNoCopy(utf8SourceInfo, (int)srcLength, false /*isCesu8*/);
+        Assert(sourceIndex != Js::Constants::InvalidSourceIndex);
+
+        Field(Js::FunctionBody*) functionBody = nullptr;
+        hr = Js::ByteCodeSerializer::DeserializeFromBuffer(
+            scriptContextUI,
+            fscrGlobalCode | fscrWillDeferFncParse | fscrCanDeferFncParse | fscrCreateParserState,
```

Comment:
Are these flags required to match what we passed to `CompileUTF8Core`? If so, it might be nice to pull this into a constant.